### ISSUE TITLE
Fromstring data type issue and solution in Python3.10 and newer versions

### ISSUE THREAD

**Issue by alikerenciler** [OpenLightingProject/ola#1987]
Hello guys,

I downloaded the newer version of OLA API. When I try to collect the DMX data with Python (py3.10), I came across with an error of:
" AttributeError: 'array.array' object has no attribute 'fromstring'"

To solve this I found 2 ways as follows.

**1. Manually arranging the files of OLA library: **
- Go to " .local/lib/python3.10/site-packages/ola/ "
- Open the " OlaClient.py " python script
- Search for "fromstring" and change them as "frombytes" (there were two function naming changed in OLA 0.10.7 lib)

**2. Downgrading Python version: ** (Not recommended)
- Create another python environment with Python 3.8.0, and there will be no problem on receiving DMX data.

Have a nice day ;)

**Comment by peternewman** [comment]
Thanks for the heads up @alikerenciler .

This was fixed here:
https://github.com/OpenLightingProject/ola/commit/3ff957f2baf0942aa82a75fffe83786dbff6e5e0

And then improved/made redundant in:
https://github.com/OpenLightingProject/ola/commit/01bd5990f9994057df42f0afc3a020f987708bd4

I assume from your comment you were running 0.10.7? As this should be in 0.10.9 onwards.

Do you want to give that version a check and confirm it works as expected for you?

**Comment by alikerenciler** [comment]
The problem relays on that the Python's newer versions have no longer fromstring data type support and I do not know the exact reason. This modification should done in OLA Library for further applications.

**Comment by alikerenciler** [comment]
Thanks for the updates @peternewman .

The OLA 0.10.9 version is not available on the PyPI webpage. https://pypi.org/project/ola/#history

Even if I try to download with pip the library yields this error of:

$ pip3 install ola==0.10.9
Defaulting to user installation because normal site-packages is not writeable
ERROR: Could not find a version that satisfies the requirement ola==0.10.9 (from versions: 0.10.7)

**Comment by guidoffm** [comment]
Would it be an option to split the big project into smaller projects and bring the python and javascript clients to own projects? In the new projects GitHub build actions could be created that also include to publish the libraries.

**Comment by peternewman** [comment]
> The OLA 0.10.9 version is not available on the PyPI webpage. https://pypi.org/project/ola/#history

Yes, please see #1451, it's an unofficial publish which hasn't been updated since. I mentioned @sul3n3t and even created https://github.com/sul3n3t/ola-python/issues/1 but didn't get a response.
 
The deb install should fix it if you're on a suitable OS, or OLA is packaged for various other package managers by various people too...

> Would it be an option to split the big project into smaller projects and bring the python and javascript clients to own projects? In the new projects GitHub build actions could be created that also include to publish the libraries.

I assume you mean the Java client, we don't really have a Javascript one yet, or not in this repo. We could in theory, but it would still need someone to actually do the releases, which they could already do from this monorepo. It would also make things more complicated, both in general as everything needs the same Protobufs to work, and things like Debian packages would then need to be built from multiple repos not just one. Plus stuff like the RDM Responder tests need the Protobufs, the RDM PIDs (shared between the OLA core and themselves) as well as the Python library.

Also I thought monorepos were all the rage at the moment? 😉 

Anyway I've closed this as a duplicate of #1451 as this specific bug has been fixed in the codebase and in a released version and the only outstanding action is releasing that version on PyPi which that issue effectively already covers.
